component/src/main/java/com/celements/photo/utilities/DefaultImageUrlExtractor.java

```diff
@@ -0,0 +1,121 @@
+package com.celements.photo.utilities;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.concurrent.Immutable;
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+
+import com.celements.model.context.ModelContext;
+import com.celements.photo.container.ImageUrlDim;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.Utils;
+
+@Immutable
+@Singleton
+@Component
+public class DefaultImageUrlExtractor implements ImageUrlExtractor {
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(DefaultImageUrlExtractor.class);
+
+  public static final int MAX_ALLOWED_DIM = 1000000000;
+  public static final int MIN_SOCIAL_MEDIA_IMAGE_SIZE = 200;
+  public static final int MIN_SOCIAL_MEDIA_AREA_SIZE = MIN_SOCIAL_MEDIA_IMAGE_SIZE
+      * MIN_SOCIAL_MEDIA_IMAGE_SIZE;
+
+  private final Pattern IMG_FROM_HTML_PATTERN = Pattern.compile("<img .*?src=['\"](.*?)['\"].*?/>");
+  private final Pattern EXTRACT_URL_PART_PATTERN = Pattern.compile("/([^/?]*)");
+  private final Pattern CELWIDTH_FROM_URL_PATTERN = Pattern.compile("^.*celwidth=(\\d+)(&.*)?$");
+  private final Pattern CELHEIGHT_FROM_URL_PATTERN = Pattern.compile("^.*celheight=(\\d+)(&.*)?$");
+
+  @Override
+  public @NotNull List<ImageUrlDim> extractImagesList(@NotNull String content) {
```

Comment:
I have issues with this service/method.
- **naming**: _extractImagesList_, but it does in fact extract image url dimensions, not images
- **return parameter**: ImageUrlDim is returned, however not every client of ImageUrlExtractor is necessarily interested in dimension. Strings or a more general ImageUrl-Object could be returned
- **reusability**: because _filterImagesByMissingOrSmallSize_ the implementation is very specific and not reusable in more general ways, e.g. if you just want to extract all image urls from a string

Some suggestions come to mind:
- have ImageUrlExtractor.extractImageUrls(String) return a list of ImageUrl-Objects
- _ImageUrl_ can be queried for a Dimension object (perhaps cached if performance heavy), also action, space, docName, etc. could be useful
- implement a guava _ImageUrlSizePredicate implements Predicate\<ImageUrl\>_ in which you can filter for missing/minimal/maximal size

WDYT?